### PR TITLE
ASR: Constant string attributes refactoring

### LINKED ISSUE #1278: String attributes need refactoring
https://github.com/lcompilers/lpython/blob/d4a6d5360ca520d9a25d658e6660cb39a2eda4e2/src/lpython/semantics/python_comptime_eval.h#L69-L76

These string methods are never used during compile-time evaluation as they are represented by ConstantStr in ASR and handled differently in `python_ast_to_asr.cpp` and so this makes the file unnecessarily lengthy as well as the `python_ast_to_asr`. Moreover, the following eval functions are also implemented incorrectly:
https://github.com/lcompilers/lpython/blob/d4a6d5360ca520d9a25d658e6660cb39a2eda4e2/src/lpython/semantics/python_comptime_eval.h#L800-L810

We should remove all the value functions from this map and replace them with `not_implemented`.

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #1278

### CODE REVIEW

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -5258,6 +5258,194 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
+                            if (j != 0) {
+                                j = lps[j - 1];
+                            } else {
+                                i = i + 1;
+                            }
+                        }
+                    }
+                }
+                tmp = ASR::make_IntegerConstant_t(al, loc, res,
+                    ASRUtils::TYPE(ASR::make_Integer_t(al,loc, 4, nullptr, 0)));
```
Comment (czgdp1807) [inline src/lpython/semantics/python_ast_to_asr.cpp:5333]: Please put this into a function and call it here.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/test_str_attributes.py [hunk 2]
```diff
@@ -67,10 +76,13 @@ def startswith():
...
-swapcase()
-find()
-startswith()
+def check():
+    capitalize()
+    lower()
+    strip()
+    swapcase()
+    find()
+    startswith()
```
Comment (namannimmo10) [inline integration_tests/test_str_attributes.py:88]: 🙂
```suggestion
    startswith()

check()
```

File: src/lpython/semantics/python_ast_to_asr.cpp (hunk 1 above)
Comment (Smit-create) [reply src/lpython/semantics/python_ast_to_asr.cpp]: Done

File: integration_tests/test_str_attributes.py (hunk 2 above)
Comment (Smit-create) [reply integration_tests/test_str_attributes.py:88]: Thanks

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**